### ISSUE TITLE
Normalize data

### ISSUE THREAD

**Issue by domist07** [cgre-aachen/gempy#320]
My data is very unequal sparsed, for example [1000, 200, 60] m and I have only point data. Orientation I calculate from the neighbour points.
For first step I transform the data with PCA:
```
from sklearn.decomposition import PCA
pca = PCA(n_components=3)
pca.fit(geo_data.surface_points.df[['X', 'Y', 'Z']])
geo_data.surface_points.df[['X', 'Y', 'Z']] = \
   pca.transform(geo_data.surface_points.df[['X', 'Y', 'Z']])
```
Second I "normalize" the data that all axes have the equal extend:
```
Min = np.array([np.min(geo_data.surface_points.df['X']),
                np.min(geo_data.surface_points.df['Y']),
                np.min(geo_data.surface_points.df['Z'])])
Max = np.array([np.max(geo_data.surface_points.df['X']),
                np.max(geo_data.surface_points.df['Y']),
                np.max(geo_data.surface_points.df['Z'])])
Diff = Max-Min

dy = Diff[0]/Diff[1]
dz = Diff[0]/Diff[2]

geo_data.surface_points.df['Y'] = geo_data.surface_points.df['Y'] * dy
geo_data.surface_points.df['Z'] = geo_data.surface_points.df['Z'] * dz
```
Then I calculate the surfaces. 
After these steps I make the obeservation that the meshs of the surfaces are much more precisely calculated to my datapoints, BUT the meshs are much more rougher. I think the problem is that the correct influence of the points to each other in the Cokriging algorithm is destroyed through my transformation and normalisation. Is it possible/Where to insert the normalisation parameters in the  Cokriging algorithm for a correct influence of the points to each other?

**Comment by javoha** [comment]
Hi domist7,

can you explain exactly what you want to achieve with those two steps? I am not knowledgeable in this field but I understand PCA is used to reduce the number input variables in general, but you are using it to transform the coordinates and keep all three, right?

Regarding the normalzation step: Kriging weights are based on distances between target location and location of input data (as well as distances between input data locations). Thus transforming the coordinates is problematic. Maybe working with anisotropy might be a solution to your problem. 

Can you maybe show an example of yout two cases and explain, what result you expect?

**Comment by domist07** [comment]
Yes your right regarding my use of PCA.

Regarding the normalzation step:
If I calculate it without normalisation I get this:
![Issue320-2](https://user-images.githubusercontent.com/59735059/76212581-1680f800-6209-11ea-91bf-92d72808b2df.png)
Smooth meshs
![Issue320-1](https://user-images.githubusercontent.com/59735059/76212580-15e86180-6209-11ea-85e6-415ae8c0d78a.png)
But points fits not good to mesh

If I calculate it with normalisation I get this:
![Issue320-3](https://user-images.githubusercontent.com/59735059/76212644-37e1e400-6209-11ea-9835-a30729aad9da.png)
Points fits very good to mesh
![Issue320-4](https://user-images.githubusercontent.com/59735059/76212646-387a7a80-6209-11ea-91da-cf28f0905818.png)
But routh meshs

For these plots I scale the Y-direction  by 2 and Z-direction by 10. Smooth factor is set for all points to zero.

What I expect is that the points are always exactly in the meshs, independent of the data sparsing. To solve this issue I had the idea with the normalisation but then I get the described problems with the Kriging.

**Comment by javoha** [comment]
Thanks, that helps alot. The smoothing parameter was also set to 0 in the first example without your "preprocessing"? If not you should try that.
If you already did that then there is something wrong on our side. The smooth factor should correspond to the so called "nugget effect" in kriging and when set to 0 the data points should be absolutely honored. We will check ot out!

**Comment by domist07** [comment]
Yes it was always set to zero.

I think the problem is the calculation acuraccy. If you show all directions with scale=1 it looks good. Only if I scale the axis you see the problem.

**Comment by javoha** [comment]
So I was able to reproduce the behavior you described: I am pretty sure its not a problem of the kriging calculations. 

Instead I think it is the way we construct the surfaces with the `skimage.measure.marching_cubes_lewiner` function. There seems to be a systematic offset, I suppose based on where the algorithm "starts".

We will take a more detailed look. For now I think if you perform any calculations on volumes, the blocks should be correct. The problem really seems to be just a visualization/surface rendering issue.

**Comment by domist07** [comment]
Ok. Thanks for the reply!
